android/java/org/mlperf/inference/RunMLPerfWorker.java

```diff
@@ -176,7 +177,9 @@ public boolean handleMessage(Message msg) {
       result.setMode(data.mode);
       callback.onBenchmarkFinished(result);
 
-      if (!backendName.contains("samsung")) {
+      //if (!backendName.contains("samsung")) {
+      Log.i(TAG, "Build.HARDWARE " + Build.HARDWARE);
+      if (!"exynos2100".equals(Build.HARDWARE)){
```

Comment:
I think we should check for both hardware and "samsung" in backend name.
If we keep only hardware check, if someone builds the app without Samsung backend and runs it on a phone with Exynos 2100, then this workaround will affect other backends.